### PR TITLE
Edsetup

### PR DESCRIPTION
**Pull request by mritv**
Edsetup lets you create your project directory and necessary files to get you up and running with on command.

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (pesser) [inline .idea/.gitignore:1]: not sure what those `.idea` files are but they probably shouldn't be commited to the repo.

File: edflow/edflow [hunk 2]
```diff
@@ -85,6 +85,10 @@ def main(opt, additional_kwargs):
     # Training
     if opt.train:
         if opt.checkpoint is not None:
+            if not os.path.exists(opt.checkpoint):
```
Comment (pesser) [inline edflow/edflow:88]: good idea but we cannot really do this: for tensorflow the specified checkpoint path is not a file

File: edflow/edsetup_files/model.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Model(object):
+    """
+    Clean model skeleton for initialization.
+    """
+
+    def __init__(self, config):
+        super().__init__()
```
Comment (pesser) [inline edflow/edsetup_files/model.py:7]: i guess we don't need this, super() is just object

File: edflow/eval/pipeline.py [hunk 4]
```diff
@@ -55,6 +55,8 @@ def eval_op(self, inputs):
 
     edflow -n myexperiment -e the_config.yaml -p path_to_project
 
+"git add/rm <conflicted_files>", then run "git rebase --continue".
```
Comment (pesser) [inline edflow/eval/pipeline.py:58]: is this supposed to be here?

File: edflow/eval/pipeline.py [hunk 5]
```diff
@@ -479,6 +481,8 @@ def save_output(root, example, index, sub_dir_keys=[], keypath="step_ops"):
     """
 
     example = retrieve(example, keypath)
+    if callable(example):
```
Comment (pesser) [inline edflow/eval/pipeline.py:484]: this shouldnt be here anymore

File: edflow/eval/pipeline.py [hunk 6]
```diff
@@ -663,6 +666,7 @@ def load_by_heuristic(path):
     elif ext == ".txt":
         return txt_loader(path)
     else:
+
```
Comment (pesser) [inline edflow/eval/pipeline.py:669]: empty line?

File: edflow/util.py [hunk 7]
```diff
@@ -137,7 +119,7 @@ def call(key, value):
 
 class KeyNotFoundError(Exception):
     def __init__(self, cause):
-        self.cause = cause
+        self.cause = casue
```
Comment (pesser) [inline edflow/util.py:122]: this is a regression to a buggy version

File: edflow/util.py [hunk 8]
```diff
@@ -41,49 +30,42 @@ def linear_var(step, start, end, start_value, end_value, clip_min=0.0, clip_max=
 
 
 def walk(dict_or_list, fn, inplace=False, pass_key=False, prev_key=""):  # noqa
-    """
-    Walk a nested list and/or dict recursively and call fn on all non
+    """Walk a nested list and/or dict recursively and call fn on all non
```
Comment (pesser) [inline edflow/util.py:33]: this is a regression to an old version with incorrect format string

File: edflow/util.py [hunk 9]
```diff
@@ -402,8 +380,7 @@ def set_value(list_or_dict, key, val, splitval="/"):
 
 
 def contains_key(nested_thing, key, splitval="/", expand=True):
-    """
-    Tests if the path like key can find an object in the nested_thing.
+    """Tests if the path like key can find an object in the nested_thing.
```
Comment (pesser) [inline edflow/util.py:383]: regression to incorrect format string

File: pip-wheel-metadata/edflow.dist-info/LICENSE [hunk 10]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (pesser) [inline pip-wheel-metadata/edflow.dist-info/LICENSE:1]: should this pip wheel metadata be commited to the repository? if so, why?

**Review by pesser — CHANGES_REQUESTED** [review]
Great to see `edsetup` coming! But currently the pull request contains some files which should not be commited, one change that is incompatible with tensorflow checkpoints and some regressions to older versions of code which have been fixed or improved in their current versions on master.
It would be best to start a new branch from the current master and add only the relevant edsetup code (`edflow/edsetup`, `edflow/edsetup_files/`, `setup.py` and `tests/test_edsetup.py`). Then we have a clean history and can be sure that we do not re-introduce problems which had been fixed already.

File: .idea/.gitignore (hunk 1 above)
Comment (mritv) [reply .idea/.gitignore:1]: Will do that.

File: edflow/edflow (hunk 2 above)
Comment (mritv) [reply edflow/edflow:88]: This is also an older commit which should not have been there. I'll create a new branch/fork of master and edit those files in there.

File: edflow/eval/pipeline.py (hunk 4 above)
Comment (mritv) [reply edflow/eval/pipeline.py:58]: Nope

**Pull request closed without merging**